### DISCUSSION TITLE
Run scheduler task on one worker

### DISCUSSION THREAD

**Discussion by salman-khandu** [fastapi/fastapi#13106] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@asynccontextmanager
async def lifespan(app: FastAPI):
    '''This function is used to run the scheduler'''
    SchedulerService.run_scheduler()
    yield
app = FastAPI( lifespan=lifespan)
```

### Description

I am running fastapi with 5 workers using gunicorn and  Here is my main.py code

`
@asynccontextmanager
async def lifespan(app: FastAPI):
    '''This function is used to run the scheduler'''
    SchedulerService.run_scheduler()
    yield
app = FastAPI( lifespan=lifespan)
`
is there best way to run scheduler only from one work ? 

### Operating System

Linux

### FastAPI Version

0.115.6

### Pydantic Version

2.9.2

### Python Version

11

**Comment by mortezaPRK** [comment]
This is not related to FastAPI, but you have generally two options:
1. Create two entrypoint for you app. one would run FastAPI using `gunicron`, the other one runs the scheduler
2. Use gunicron hooks and define a `env` for the first worker. Then in your code, only run the scheduler if that env is set. this might be helpful https://stackoverflow.com/a/24101801

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.

Relevant: https://github.com/fastapi/fastapi/discussions/12793
